lib/libxdp/libxdp.c

```diff
@@ -2119,9 +2221,14 @@ static struct xdp_multiprog *xdp_multiprog__generate(struct xdp_program **progs,
 			goto err;
 	}
 
+	if (num_progs > num_new_progs)
```

Comment:
Think the check should be if (old_mp), so it will cover the case where old_mp->num_links == 0. I think this can happen if the program loaded is a hw offload case only.

Reply:
Looks like you fixed it in the next patch :)